awx/main/tasks.py

```diff
@@ -2393,6 +2410,14 @@ def post_run_hook(self, instance, status):
             if status == 'successful' and instance.launch_type != 'sync':
                 self._update_dependent_inventories(instance, dependent_inventory_sources)
 
+        # Clean up the oauth token if we are project syncing
+        if instance.project.sync_assets:
+            try:
+                OAuth2AccessToken.objects.filter(user=instance.created_by_id, description='project sync {}'.format(instance.id)).delete()
+            except Exception as e:
+                pass
```

Comment:
If we fail to clean up the token for some reason, it's probably worth logging:

```python
logger.exception("Failed to cleanup up OAuth2.0 token used for asset syncing")
```

Reply:
Added a message for both creation and deletion of tokens.